src/traces/contour/plot.js

```diff
@@ -291,18 +313,303 @@ function makeLines(plotgroup, pathinfo, contours) {
+                    type: 'linear',
+                    _separators: '.,',
+                    _id: 'ycontour',
+                    nticks: pathinfo.length,
+                    showexponent: 'all',
+                    range: [contours.start, contours.end]
+                };
+                setConvert(formatAxis, gd._fullLayout);
+                Axes.calcTicks(formatAxis);
+                formatAxis._tmin = null;
+                formatAxis._tmax = null;
+            }
+            contourFormat = function(v) {
+                return Axes.tickText(formatAxis, v).text;
+            };
+        }
+
+        var dummyText = defs.append('text')
+            .attr('data-notex', 1)
+            .call(Drawing.font, contours.font);
+
+        var xLen = pathinfo[0].xaxis._length;
+        var yLen = pathinfo[0].yaxis._length;
+
+        // visible bounds of the contour trace (and the midpoints, to
+        // help with cost calculations)
+        var bounds = {
+            left: Math.max(perimeter[0][0], 0),
+            right: Math.min(perimeter[2][0], xLen),
+            top: Math.max(perimeter[0][1], 0),
+            bottom: Math.min(perimeter[2][1], yLen)
+        };
+        bounds.middle = (bounds.top + bounds.bottom) / 2;
+        bounds.center = (bounds.left + bounds.right) / 2;
+
+        var plotDiagonal = Math.sqrt(xLen * xLen + yLen * yLen);
+
+        // the path length to use to scale the number of labels to draw:
+        var normLength = constants.LABELDISTANCE * plotDiagonal /
+            Math.max(1, pathinfo.length / constants.LABELINCREASE);
+
+        linegroup.each(function(d) {
+            // - make a dummy label for this level and calc its bbox
+            var text = contourFormat(d.level);
+            dummyText.text(text)
+                .call(svgTextUtils.convertToTspans, gd);
+            var bBox = Drawing.bBox(dummyText.node());
+            var textWidth = bBox.width;
+            var textHeight = bBox.height;
+            var dy = (bBox.top + bBox.bottom) / 2;
+            var textOpts = {
+                text: text,
+                width: textWidth,
+                height: textHeight,
+                level: d.level,
+                dy: dy
+            };
+
+            d3.select(this).selectAll('path').each(function() {
+                var path = this;
+                var pathBounds = Lib.getVisibleSegment(path, bounds, textHeight / 2);
+                if(!pathBounds) return;
+
+                var onPlotMin = pathBounds.min;
+                var onPlotMax = pathBounds.max;
+                var totalPathLen = pathBounds.total;
+                var pathLen = onPlotMax - onPlotMin;
+
+                var isOpen = d3.select(this).classed('openline');
+
+                if(pathLen < textWidth * constants.LABELMIN) return;
+
+                var maxLabels = Math.min(Math.ceil(pathLen / normLength),
+                    constants.LABELMAX);
+                var dp, p0, pMax, minCost, location, pMin;
```

Comment:
*Only* danger (I think) is is `var` declaration gets removed. Linter may or may not catch it. So really not dangerous at all… almost certainly 🔫